pynicotine/gtkgui/search.py

```diff
@@ -1415,19 +1415,11 @@ def on_download_folders(self, *_args, download_location=""):
                 visible_files.append(
```

Comment:
Could you add these files from the search results to the dialog when showing it? In the past, some users wanted to download all visible files in a folder of an offline user, and I'd like to keep this functionality.

Reply:
One of the benefits would be that the list could already be partially populated with the originally visible items while waiting for the entire folder contents response. This is relevant if the peer connection is very slow and/or the data is large (it might take several minutes).

Hence, there should not be anything to stop the Download button from being pressed if we can't be bothered to wait (or if the peer just went offline). Then at least we would be able get the files we already know about into the queue.